File: webApp/webapi/Models/Book.cs
﻿using Newtonsoft.Json;

namespace webapi.Models
{
    public class Book
    {
        public Guid ID { get; set; }
        public string? AaName { get; set; }
        public string? Author { get; set; }
        public string? Series { get; set; }
        public bool OwnedOrWish { get; set; }
        public Status Status { get; set; }
        public Score Score { get; set; }
        public string? Comment { get; set; }
        [JsonIgnore]
        public virtual List<BookTrope> BookTropes { get; set; } = new();
        [JsonIgnore]
        public virtual List<BookGenre> BookGenres { get; set; } = new();
    }
    public enum Status
    {
        NotStarted,
        CurrentlyReading,
        Read,
        WantToReadSoon
    }
    public enum Score
    {
        One,
        Two,
        Three,
        Four,
        Five
    }
}

File: webApp/angularapp/src/app/shared/models/Book.ts
import { Genre } from './Genre';
import { Trope } from './Trope';

export interface Book {
  aaName: string;
  author: string;
  series: string;
  ownedOrWish: boolean;
  status: Status;
  score: Score;
  comment: string;
  tropes: Trope[] | undefined | null;
  genres: Genre[] | undefined | null;
}

export enum Status {
  NotStarted,
  CurrentlyReading,
  Read,
  WantToReadSoon,
}

export enum Score {
  One,
  Two,
  Three,
  Four,
  Five,
}

export interface StatusDisplay {
  key: Status;
  value: string;
}

export interface ScoreDisplay {
  key: Score;
  value: string;
}
